source/Ship.cpp

```diff
@@ -807,7 +824,9 @@ string Ship::FlightCheck() const
 		if(fuelCapacity < JumpFuel())
 			return "no fuel?";
 	}
-	
+	if((generation + battery + solar) < maxWeaponEnergy)
```

Comment:
Note that the total energy available on a given frame is calculated already, as the local variable `energy`. (It comprises normal generation, storage capacity, solar collection, and fuel->energy conversion.)

So really you could distill the check to just:
```
...
for(const auto &it : outfits)
  if(it.first->FiringEnergy() > energy)
    return "insufficient energy to fire?";
```
(All `Outfit`s derive from the `Weapon` base class, so they all have [`FiringEnergy`](https://github.com/endless-sky/endless-sky/blob/6072a633dc87a0dfcf851825d3d3df6770e96785/source/Weapon.h#L180) available, with the default value of `0.0`.)
This isn't performance-critical code, so there isn't any penalty to keeping the expressiveness, i.e. `if(it.first->IsWeapon() && it.first->FiringEnergy() > energy)` (in case we ever create a non-`Weapon` use for `FiringEnergy()`).

Reply:
My bad! I didn't know fuel->energy conversion was a thing. It just goes to show I should read more of the source code.

I've removed the MaxWeaponEnergy function in favor of your inline version, except I've kept the check for IsWeapon() for easier readability. If we ever need to know the max weapon energy elsewhere, we can refactor this.